pkg/provisioner/ironic/ironic.go

```diff
@@ -1919,6 +1919,16 @@ func (p *ironicProvisioner) IsReady() (result bool, err error) {
 }
 
 func (p *ironicProvisioner) HasCapacity() (result bool, err error) {
+	bmcAccess, err := p.bmcAccess()
+	if err != nil {
+		return false, err // shouldn't actually happen so late in the process
+	}
+
+	// If the current host uses virtual media, do not limit it. Virtual
+	// media deployments may work without DHCP and can share the same image.
+	if bmcAccess.SupportsISOPreprovisioningImage() {
+		return true, nil
+	}
```

Comment:
I understand that this could be an handy centralized place to perform this check, and it may sound as a reasonable location froma technical point of view. Anyhow I'd evaluate also the possibility to "push it up" in the stack - where the `ensureCapacity` is invoked, essentially in the `updateHostStateFrom` and `checkDelayedHost`. Not to save any comp cycle, but just to move such logic, which is strictly host dependent, out from the provisioner - so that the opt out decision could be made earlier

Reply:
I'm not sure I'm following, this logic is also provisioner-dependent, it relies on knowing how Ironic works. We'd also need to move the filtering of the busy hosts up the stack somehow, it's going to be a significant change.

Reply:
I was thinking on a more general mechanism to avoid applying the delay check mechanism at all based on some predicate (in this case it'd be based on `bmcAccess.SupportsISOPreprovisioningImage()`, but in future it could be something else, maybe an host label) (btw, using a label wouldn't need to recompile necessary the code in case a new scenario will pop up).
I won't block on this, but I'd consider a more generic mechanism in future if other scenarios will arise